camelConnector/src/main/java/io/vantiq/extsrc/camelconn/discover/CamelResolver.java

```diff
@@ -188,50 +204,123 @@ Collection<File> resolve(String organization, String name, String revision, Stri
-        //init resolve report
+        // There are some dependencies that don't seem to resolve correctly, at least with our implementation.  To
+        // counter this, we track a set of places where we need to override the type provided by the maven repo.
+        // This situation typically arises when the artifact in question specifies a packaging attribute for some
+        // configurations, and the configuration for which we're locking qualifies.  To manage this, we'll add a
+        // DependencyDescriptorMediator (via the lambda below) which, when it sees the dependency specified in the
+        // typeOverrides list, will add a DependencyArtifact to force the downloading of th correct type/extension.
+        // these adjustments allow us to include having to special case lots of stuff in the kamelet/assembly
+        // specifications.  It also will provide the same functions if users configure their own camelConnectors that
+        // encounter these situations.
+        
+        typeOverrides.forEach( (src, newType) -> {
+            String[] parts = src.split(":");
+            DependencyDescriptorMediator ddm = dependencyDescriptor -> {
+                String targetOrg = dependencyDescriptor.getDependencyRevisionId().getOrganisation();
+                String targetName = dependencyDescriptor.getDependencyRevisionId().getName();
+                String targetRev = dependencyDescriptor.getDependencyRevisionId().getRevision();
+                log.trace("CamelConn mediator called with {}:{}:{}", targetOrg, targetName, targetRev);
+                // We should get called only with the correct set, but I have seen things behave differently, so
+                // better to be safe.  So here we check that the passed-in dependency (target*) match the type
+                // overrides for which this DependencyDescriptorMediator is being constructed.
+                if (parts[0].equals(targetOrg) && parts[1].equals(targetName) && parts[2].equals(targetRev)) {
+                    log.trace(">>>> Specifying artifact for {}:{}:{}:{}", targetOrg, targetName, targetRev,
+                              newType);
+                    DependencyArtifactDescriptor dad =
+                            new DefaultDependencyArtifactDescriptor(dependencyDescriptor, targetName, newType,
+                                                                    newType, null, null);
+                    if (dependencyDescriptor instanceof DefaultDependencyDescriptor) {
+                        // PomModuleDescriptorBuilder.PomDependencyDescriptor is based on
+                        // DefaultDependencyDescriptor so this covers that case as well.
+                        
+                        // In this case, just update the existing dependency
+                        ((DefaultDependencyDescriptor)
+                                dependencyDescriptor).addDependencyArtifact(ANY_CONFIGURATION, dad);
+                    } else {
```

Comment:
at present, the only dependency descriptor type is that shown above. But we'll want to future proof things _just in case_.  We log an error & try to recover...